src/components/RangeSlider/components/SingleThumb/SingleThumb.scss

```diff
@@ -114,51 +173,38 @@ $range-thumb-shadow-focus: 0 0 0 rem(1px) color('indigo');
     transform: translateY($range-thumb-size / 5) scale(0.4);
   }
 
-  &::-webkit-slider-thumb {
-    margin-top: -($range-thumb-size - $range-track-height) / 2;
-  }
-
   .error & {
     --progress-lower: #{color('red')};
 
     @include range-track-selectors {
-      background-color: color('red', 'light');
+      background-color: var(--p-override-none, color('red', 'light'));
     }
 
     @include range-thumb-selectors {
-      border-color: color('red');
+      background: var(--p-critical, color('white'));
+      border-color: var(--p-critical, color('red'));
       box-shadow: $range-thumb-shadow-error;
     }
   }
 
   .disabled & {
     @include range-track-selectors {
       cursor: auto;
+      background-color: var(--p-border-on-inverse, color('sky', 'dark'));
       background-image: none;
     }
 
     @include range-thumb-selectors {
       cursor: not-allowed;
-      border-color: color('sky', 'dark');
-    }
-  }
-
-  // match `error` specificity for interaction styles
-  .RangeSlider &:focus {
-    // repeated so that `focus` can override `error`
-    --progress-lower: #{color('indigo')};
-
-    @include range-track-selectors {
-      background-color: color('sky', 'dark');
-    }
-
-    @include range-thumb-selectors {
-      background: linear-gradient(
-        color('sky', 'lighter'),
-        color('sky', 'light')
-      );
-      border-color: color('indigo');
-      box-shadow: $range-thumb-shadow-focus;
+      border-color: var(--p-border-on-surface, color('sky', 'dark'));
+      background: var(--p-surface-background, color('white'));
+
+      &:hover {
+        background: linear-gradient(
+          var(--p-surface-background, color('sky', 'lighter')),
+          var(--p-surface-background, color('sky', 'light'))
+        );
+      }
```

Comment:
I would expect the disabled range slider to use `interactiveActionDisabled` for the handle and applied portion of the track